cloudinit/config/cc_rightscale_userdata.py

```diff
@@ -1,130 +0,0 @@
-# Copyright (C) 2011 Canonical Ltd.
-# Copyright (C) 2012, 2013 Hewlett-Packard Development Company, L.P.
-#
-# Author: Scott Moser <scott.moser@canonical.com>
-# Author: Juerg Haefliger <juerg.haefliger@hp.com>
-#
-# This file is part of cloud-init. See LICENSE file for license information.
-
-import logging
-import os
-from urllib.parse import parse_qs
-
-from cloudinit import url_helper as uhelp
-from cloudinit import util
-from cloudinit.cloud import Cloud
-from cloudinit.config import Config
-from cloudinit.config.schema import MetaSchema, get_meta_doc
-from cloudinit.distros import ALL_DISTROS
-from cloudinit.settings import PER_INSTANCE
-
-MY_NAME = "cc_rightscale_userdata"
-MY_HOOKNAME = "CLOUD_INIT_REMOTE_HOOK"
-
-"""Rightscale Userdata: Support rightscale configuration hooks"""
-
-MODULE_DESCRIPTION = """\
-This module adds support for RightScale configuration hooks to cloud-init.
```

Comment:
This sounds a lot like an attempt to create a datasource inside of a configuration module. I'm curious why it was never done that way in the first place.